native/frontend/src/org/jetbrains/kotlin/resolve/konan/diagnostics/DefaultErrorMessagesNative.kt

```diff
@@ -38,7 +38,14 @@ private val DIAGNOSTIC_FACTORY_TO_RENDERER by lazy {
             "@ThreadLocal is applicable only to property with backing field, to property with delegation or to objects"
         )
         put(ErrorsNative.INAPPLICABLE_THREAD_LOCAL_TOP_LEVEL, "@ThreadLocal is applicable only to top level declarations")
-        put(ErrorsNative.INVALID_CHARACTERS_NATIVE, "Name {0}", CommonRenderers.STRING);
+        put(ErrorsNative.INVALID_CHARACTERS_NATIVE, "Name {0}", CommonRenderers.STRING)
+        put(ErrorsNative.REDUNDANT_SWIFT_REFINEMENT, "An ObjC refined declaration can't also be refined in Swift")
+        put(
+            ErrorsNative.INCOMPATIBLE_OBJC_REFINEMENT_OVERRIDE,
+            "Refined declaration overrides non refined declarations from {0}",
```

Comment:
The message is probably not clear enough for different kinds of refinements, e.g. if super method is refined in Swift and this method is refined for Obj-C.
It should be enough to simply clarify the language.